shared/middleware/files_api.rb

```diff
@@ -348,7 +347,10 @@ def put_file(endpoint, encrypted_channel_id, filename, body)
     tab_id = params['tabId']
     conflict unless buckets.check_current_version(encrypted_channel_id, filename, current_version, should_replace, timestamp, tab_id, current_user_id)
 
-    response = buckets.create_or_replace(encrypted_channel_id, filename, body, version_to_replace)
+    storage_apps = StorageApps.new(get_storage_id)
```

Comment:
It seems strange to pass in the owner's storage ID here, in fact it looks like `get_abuse` won't even use that storage ID at all. I would suggest making `get_abuse` into a class method and just calling it as `StorageApps.get_abuse(...)` below and wherever else it is called. to do this you will need to use `PEGASUS_DB[:storage_apps]` instead of `@table` within `get_abuse`, as is done in this example: https://github.com/code-dot-org/code-dot-org/blob/c7f883639668e7074cb0ed5227e7e09e3fa0622a/shared/middleware/helpers/storage_apps.rb#L323